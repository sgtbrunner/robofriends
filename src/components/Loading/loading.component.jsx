import React from 'react';
import './loading.styles.css';

const Loading = () => <h1 className="t1 loading" style={{ fontSize: '4rem' }}>Loading</h1>;

export default Loading;
