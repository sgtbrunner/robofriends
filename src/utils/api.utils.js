export const USERS_URL = 'https://jsonplaceholder.typicode.com/users';

export default {
  getUsers: () =>
    fetch(USERS_URL).then((response) => {
      if (response.ok) return response.json();
      throw new Error('Something went wrong');
    }),
};
