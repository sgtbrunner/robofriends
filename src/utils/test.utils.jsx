import React from 'react';
import { render } from '@testing-library/react';

export const mountComponent = ({ component, props = {} }) => {
  const Component = component;
  return render(<Component {...props} />);
};

export default null;
