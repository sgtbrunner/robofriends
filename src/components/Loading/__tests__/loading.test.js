import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Loading from '../index';

describe('Loading component', () => {
  test('should match snapshot', () => {
    const { container } = mountComponent({ component: Loading });
    expect(container).toMatchSnapshot();
  });

  test('check component loading text', () => {
    mountComponent({ component: Loading });
    const loading = screen.getByRole('heading', { name: /loading/i });
    expect(loading).toBeInTheDocument();
  });
});
