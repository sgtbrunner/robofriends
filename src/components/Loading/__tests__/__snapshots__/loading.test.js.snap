// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should match snapshot 1`] = `
<div>
  <h1
    class="t1 loading"
    style="font-size: 4rem;"
  >
    Loading
  </h1>
</div>
`;
